### PR TITLE
js-md5 - add ArrayBuffer parameter

### PR DESCRIPTION
**Pull request by mwmccarthy**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/emn178/js-md5#example>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by mwmccarthy** [comment]
There are some other changes I wanted to make to this also because I'm not sure what's going on, but I wanted to check first to see what I might be missing. It looks like these definitions expect the String prototype to have been modified in some way? And what does this module have to do with JQuery?

**Comment by DanielRosenwasser** [comment]
@tigerxy any idea why were these overloads originally added? The source code doesn't seem to indicate that JQuery or String's prototype get modified.

@emn178 do you know if this was ever the behavior of js-md5? Sorry for pestering you.

**Comment by emn178** [comment]
jQuery and prototype methods are not included in js-md5. Please remove.

**Comment by DanielRosenwasser** [comment]
Thanks a bunch for the response @emn178!

So @mwmccarthy, feel free to remove the augmentations and remove the `/// <types ...>`.

Then, I'd wrap the interfaces in a

```ts
declare namespace md5 {
    // ... interfaces go here
}
```

and replace the

```ts
declare module "js-md5" {
     export = md5;
 }
declare var md5: md5;
```

with the following:

```tsd
declare var md5: md5.md5;
export = md5;
```

Currently this `.d.ts` is not super accurate with the library itself. I think we'd welcome a rewrite.

**Comment by mwmccarthy** [comment]
@DanielRosenwasser, thanks I'll rewrite it as you suggest. I appreciate the feedback.

File: js-md5/js-md5-tests.ts [hunk 1]
```diff
@@ -1,20 +1,29 @@
+import * as md5 from "js-md5";
```
Comment (DanielRosenwasser) [inline js-md5/js-md5-tests.ts]: Last update - can you switch this to `import md5 = require("md5")`

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: js-md5/js-md5-tests.ts (hunk 1 above)
Comment (mwmccarthy) [inline js-md5/js-md5-tests.ts]: Absolutely. Should I do something with the version number too? It looks like the travis build doesn't like it for some reason.

Comment (DanielRosenwasser) [inline js-md5/js-md5-tests.ts]: Yes! Didn't catch that, can you switch that to v0.4 if that's the correct version?

Comment (mwmccarthy) [inline js-md5/js-md5-tests.ts]: Should be good to go.

**Pull request merged**